plugin/path_config.go

```diff
@@ -133,18 +165,18 @@ func (b *GcpAuthBackend) pathConfigRead(ctx context.Context, req *logical.Reques
 // config reads the backend's gcpConfig from storage.
 // This assumes the caller has already obtained the backend's config lock.
 func (b *GcpAuthBackend) config(ctx context.Context, s logical.Storage) (*gcpConfig, error) {
-	config := &gcpConfig{}
+	config := &gcpConfig{
+		GCEAuthMetadata: authmetadata.NewHandler(gceAuthMetadataFields),
+		IAMAuthMetadata: authmetadata.NewHandler(iamAuthMetadataFields),
+	}
 	entry, err := s.Get(ctx, "config")
-
 	if err != nil {
 		return nil, err
 	}
-	if entry == nil {
-		return nil, nil
-	}
-
-	if err := entry.DecodeJSON(config); err != nil {
-		return nil, err
+	if entry != nil {
+		if err := entry.DecodeJSON(config); err != nil {
```

Comment:
Why move this into a nested structure?

Reply:
I did it because previously, if the entry was nil, it returned so we never needed to worry about a panic on this line. However, on this branch I edited a config to always be returned so that the config would always have the new handlers on it, avoiding any risk of panics.

Reply:
I recommend inverting the conditional it as it makes it easier to follow and keeps the happy path towards the left:
```go
if entry == nil {
  return config, nil
}
err = entry.DecodeJSON(config)
if err != nil {
  return nil, err
}
return config, nil
```

Reply:
Can do! https://github.com/hashicorp/vault-plugin-auth-gcp/pull/92/commits/d96be7888a88a611ffb68f1bdcd20f09386c9837